apm-agent-core/src/test/java/co/elastic/apm/agent/impl/BinaryHeaderMapAccessor.java

```diff
@@ -48,10 +49,12 @@ private BinaryHeaderMapAccessor() {
         return headerMap.get(headerName);
     }
 
-    @Nullable
     @Override
-    public Iterable<byte[]> getHeaders(String headerName, Map<String, byte[]> headerMap) {
-        return List.of(headerMap.get(headerName));
+    public <S> void forEach(String headerName, Map<String, byte[]> carrier, S state, HeaderConsumer<byte[], S> consumer) {
```

Comment:
I wonder if multi-headers is even a thing for binary. When implementing binary `tracestate` it could make things more complicated. But we don't have to decide now.